components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/dao/CibaMgtDAOImpl.java

```diff
@@ -0,0 +1,418 @@
+
+                prepStmt.setString(1, authenticatedUser.getUserName());
+                prepStmt.setString(2, authenticatedUser.getUserStoreDomain());
+                prepStmt.setInt(3, tenantID);
+                prepStmt.setInt(4, idpID);
+                prepStmt.setString(5, AuthReqStatus.AUTHENTICATED.toString());
+                prepStmt.setString(6, key);
+                prepStmt.execute();
+                IdentityDatabaseUtil.commitTransaction(connection);
+                if (log.isDebugEnabled()) {
+                    log.debug("Successfully persisted the successful authentication identified by AuthCodeDOKey: " +
+                            key);
+                }
+            } catch (SQLException e) {
+                IdentityDatabaseUtil.rollbackTransaction(connection);
+                throw new CibaCoreException("Error occurred in persisting the successful authentication.", e);
+            }
+        } catch (SQLException e) {
+            throw new CibaCoreException("Error occurred in persisting the successful authentication.", e);
+        }
+    }
+
+    @Override
+    public boolean isAuthReqIdExist(String authReqId) throws CibaCoreException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.CibaSQLQueries.
+                    CHECK_IF_AUTH_REQ_ID_EXISTS)) {
+
+                prepStmt.setString(1, authReqId);
+                ResultSet resultSet = prepStmt.executeQuery();
+
+                if (resultSet.next()) {
+                    //do nothing
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully checked whether provided auth_req_id : " + authReqId +
+                                " exists.Provided AuthReqId exists.It is from a valid auth_req_id.");
+                    }
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        } catch (SQLException e) {
+            throw new CibaCoreException("Unsuccessful in checking whether provided auth_req_id: " + authReqId +
+                    " exists.", e);
+        }
+    }
+
+    @Override
+    public String getCibaAuthCodeKey(String authReqId) throws CibaCoreException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.CibaSQLQueries.
+                    RETRIEVE_CIBA_AUTH_CODE_KEY_BY_CIBA_AUTH_REQ_ID)) {
+
+                prepStmt.setString(1, authReqId);
+                ResultSet resultSet = prepStmt.executeQuery();
+                if (resultSet.next()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully returning CibaAuthCodeKey : " + resultSet.getString(1) + "for the " +
+                                "ciba auth_req_id : " + authReqId);
```

Comment:
auth_req_id 

is this a request param name?

Reply:
yes...the param input is auth_req_id